modules/transaction-payment/src/lib.rs

```diff
@@ -1391,21 +1406,32 @@ impl<T: Config + Send + Sync> TransactionPayment<T::AccountId, PalletBalanceOf<T
 where
 	PalletBalanceOf<T>: Send + Sync + FixedPointOperand,
 {
-	fn reserve_fee(who: &T::AccountId, weight: Weight) -> Result<PalletBalanceOf<T>, DispatchError> {
-		let fee = Pallet::<T>::weight_to_fee(weight);
-		<T as Config>::Currency::reserve_named(&RESERVE_ID, who, fee)?;
+	fn reserve_fee(
+		who: &T::AccountId,
+		fee: PalletBalanceOf<T>,
+		named: Option<ReserveIdentifier>,
+	) -> Result<PalletBalanceOf<T>, DispatchError> {
+		let fee = Pallet::<T>::calculate_final_fee(fee);
```

Comment:
if the method is passing fee directly, then we shouldn't modify it with multiplier otherwise it will be confusing
also the storage deposit is not subject to multiplier, only the weight fee is

Reply:
add a `fee_multiplier` flag as parameter, for storage fee, passing it as false, while other case may set it to true.

Reply:
add a `Multiplier` parameter to `calculate_final_fee` of Pallet, invoker should invoke this if they want to apply multiplier. Notice, if None is provide, then it will use `next_fee_multiplier` from onchain storage.